lib/index.js

```diff
@@ -50,29 +52,40 @@ function addHook(hook, opts) {
   exts.forEach(function registerExtension(ext) {
     var oldLoader;
     if (typeof ext !== 'string') throw new TypeError('Invalid Extension: ' + ext);
-    oldLoader = Module._extensions[ext] || originalJSLoader;
+    oldLoaders[ext] = oldLoader = Module._extensions[ext] || originalJSLoader;
 
-    Module._extensions[ext] = function newLoader(mod, filename) {
+    loaders[ext] = Module._extensions[ext] = function newLoader(mod, filename) {
       var compile;
       if (!reverted) {
-        compile = mod._compile;
-        mod._compile = function fakeCompile(code /*, filename*/) {
-          var newCode = code;
-
-          if (shouldCompile(filename, exts, matcher, ignoreNodeModules)) {
+        if (shouldCompile(filename, exts, matcher, ignoreNodeModules)) {
+          compile = mod._compile;
+          mod._compile = function _compile(code /*, filename*/) {
+            var newCode;
+            mod._compile = compile;
```

Comment:
I'm pretty sure this isn't what we want to do, because then the hook will only work once. (It will remove itself.)